src/overlay/Peer.cpp

```diff
@@ -1868,7 +2007,167 @@ Peer::PeerMetrics::PeerMetrics(VirtualClock::time_point connectedTime)
     , mDuplicateFloodMessageRecv(0)
     , mUniqueFetchMessageRecv(0)
     , mDuplicateFetchMessageRecv(0)
+    , mTxHashReceived(0)
     , mConnectedTime(connectedTime)
+    , mMessagesFulfilled(0)
+    , mBannedMessageUnfulfilled(0)
+    , mUnknownMessageUnfulfilled(0)
 {
 }
+
+bool
+Peer::isPullModeEnabled() const
+{
+    return mPullModeEnabled;
+}
+
+size_t
+Peer::getMaxAdvertSize() const
+{
+    auto const& cfg = mApp.getConfig();
+    double ledgerCloseTime =
+        std::chrono::duration_cast<std::chrono::milliseconds>(
+            cfg.getExpectedLedgerCloseTime())
+            .count();
+    double advertTime = cfg.FLOOD_ADVERT_PERIOD_MS.count();
+    double ledgerSize = mApp.getLedgerManager().getLastMaxTxSetSizeOps() *
+                        cfg.FLOOD_OP_RATE_PER_LEDGER;
+
+    size_t res = std::ceil(ledgerSize * advertTime / ledgerCloseTime);
```

Comment:
please do not do division with doubles and conversion to size_t like this, as it causes overflow and undefined behavior. Use bigDivideOrThrow instead, see TransactionQueue::getMaxOpsToFloodThisPeriod() as an example.

Reply:
Fixed.